adr/2025_06_13-extensions.md

```diff
@@ -0,0 +1,49 @@
+# BlockNote Extension API
+
+It is definitely more designed by accident than intention, so let's put some thought into the sort of API we would want people to build extensions with.
+
+## Core Requirements
+
+What I identified was:
+
+- The ability to "eject" to the prosemirror API, if we don't provide something good enough. This is a core choice, and one I don't think we would ever need to walk back on. Fundamentally, we do not want to expose absolutely everything that Prosemirror can do, but also do not want to stop those with the know-how to actually get stuff done.
+- A unified store API, to make consuming extension state homogenous, this will be discussed further below
+- Life-cycle event handlers, by providing hooks for `create`, `mount`, and `unmount` we allow extensions to attach event handlers to DOM elements, and have a better general understanding of the current state of the editor. The `onCreate` handler will be very handy to guarantee access to the editor instance before anything is called.
+- Editing event handlers, by providing hooks for `change`, `selectionChange`, `beforeChange`, and `transaction` we give extensions access to the fundamental primitives of the editor.
+
+## State Management
+
+What I had the most trepidation about was deciding on whether we should prescribe a state management library, and if so, which one.
+
+I think the answer is _yes_, we should prescribe a state management library, and I think the answer is @tanstack/store.
```

Comment:
I think the best way to validate this is to create a PR that migrates the AI extension from zustand to tanstack. Let's create a task for this?